ldm/gfpgan/gfpgan_tools.py

```diff
@@ -75,52 +75,50 @@ def run_gfpgan(image, strength, seed, upsampler_scale=4):
 
 def _load_gfpgan_bg_upsampler(bg_upsampler, upsampler_scale, bg_tile=400):
     if bg_upsampler == 'realesrgan':
-        if not torch.cuda.is_available():  # CPU
-            warnings.warn(
-                'The unoptimized RealESRGAN is slow on CPU. We do not use it. '
-                'If you really want to use it, please modify the corresponding codes.'
-            )
-            bg_upsampler = None
+        if not torch.cuda.is_available(): # CPU or MPS on M1
+            use_half_precision = False
```

Comment:
You do not need this condition. Instead in `RealESRGANer()` you can simply replace `half` attribute by `half=torch.cuda.is_available()`

Reply:
Absolutely agreed, I just intended to have a minimal change and keep the intention clear. E.g. we keep the "# CPU / MPS" comment and keep the "use_half_precision" as label of what is done. Mainly was afraid that there might be other cases here apart from MPS and CUDA that might need a different code path here.